### PR TITLE
Removed dynamic model.choice on Project.localfiletype

### LINKED ISSUE #4110: Using dynamic choices on model causes migration issues
There are some dynamic choice on models - if a deployment env is set up differently to the env in which the last migration was created - it causes makemigrations to want to create a new migration.

See here:

https://github.com/translate/pootle/blob/master/pootle/apps/pootle_project/models.py#L159:L160

Generally choices shouldnt be dynamic in this way:

https://docs.djangoproject.com/en/dev/ref/models/fields/#choices

See also #3687

### PR DESCRIPTION
**Pull request by phlax**
Removed the model choices as they were causing migration issues.

Added check in Manager.create and Model.save to ensure localfiletype
is valid

Fixes #4110

### CODE REVIEW

**Comment by unho** [comment]
Reviewed and tested locally. gtm from my side. Please wait for another gtm.

**Comment by julen** [comment]
Not defining a `choices` means automatic validation is left behind. Keeping that in mind, this looks like a perfect match for [Django 1.8's `forms.ChoiceField` accepting a callable for `choices`](https://github.com/django/django/commit/74e1980cf96eb45079bef464fabdcbe0a6db2423). While one could potentially insert invalid data by using the ORM, `localfiletype` is only exposed via forms, where validation happens.

**Comment by phlax** [comment]
> Not defining a choices means automatic validation is left behind

yep, thats why i added the create/save validation. The only problem i see with this approach is that it doesnt validate on update.

It defo should have validation on any forms too.

**Comment by julen** [comment]
My point is you don't need to introduce any manual validation code, but ensure the form knows about the dynamic choices.

**Comment by unho** [comment]
IMHO the best is to validate both in the model and the form. Django does this automatically for you if you specify the choices in the model, so I think it is better to keep that behavior. Also doing the validation just in the form might be problematic in case the model data does not come from the form.

**Comment by phlax** [comment]
> ... ensure the form knows about the dynamic choices.

@julen ive added form validation also - are you happy for this to land?

File: pootle/apps/pootle_app/forms.py [hunk 1]
```diff
@@ -16,9 +16,11 @@
 
 from pootle_language.models import Language
 from pootle_project.models import Project, RESERVED_PROJECT_CODES
+from pootle_store.filetypes import filetype_choices
 from pootle_store.models import Store
 
 
+
```
Comment (unho) [inline pootle/apps/pootle_app/forms.py]: Unnecessary blank line. Please remove.

**Comment by julen** [comment]
It's enough to validate the form as I already mentioned in the previous comment.

**Comment by phlax** [comment]
>  It's enough to validate the form as I already mentioned in the previous comment.

What happens if the Project is created programatically?

**Comment by julen** [comment]
> What happens if the Project is created programatically?

[While one could potentially insert invalid data by using the ORM, localfiletype is only exposed via forms, where validation happens.](https://github.com/translate/pootle/pull/4122#issuecomment-145495244)

**Comment by julen** [comment]
Looks good to me.

**Pull request merged**